js/src/redux/middleware.js

```diff
@@ -25,24 +25,38 @@ import CertificationsMiddleware from './providers/certifications/middleware';
 import ChainMiddleware from './providers/chainMiddleware';
 import RegistryMiddleware from './providers/registry/middleware';
 
-export default function (api, browserHistory) {
-  const errors = new ErrorsMiddleware();
-  const signer = new SignerMiddleware(api);
-  const settings = new SettingsMiddleware();
-  const status = statusMiddleware();
-  const certifications = new CertificationsMiddleware();
-  const routeMiddleware = routerMiddleware(browserHistory);
-  const chain = new ChainMiddleware();
-  const registry = new RegistryMiddleware(api);
-
-  const middleware = [
-    settings.toMiddleware(),
-    signer.toMiddleware(),
-    errors.toMiddleware(),
-    certifications.toMiddleware(),
-    chain.toMiddleware(),
-    registry
-  ];
+export default function (api, browserHistory, forEmbed = false) {
+  let middleware;
+  if (forEmbed) {
+    const errors = new ErrorsMiddleware();
```

Comment:
Move the common middle out of the `if`/`else` blocks. Use `concat` to add middleware.